test/t/io/test_writer.cpp

```diff
@@ -111,3 +111,56 @@ TEST_CASE("Writer") {
 
 }
 
+TEST_CASE("Writer (user-provided pool)") {
+    osmium::io::Header header;
+    header.set("generator", "test_writer.cpp");
+
+    osmium::io::Reader reader{with_data_dir("t/io/data.osm")};
+    osmium::memory::Buffer buffer = reader.read();
+    REQUIRE(buffer);
+    REQUIRE(buffer.committed() > 0);
+    const auto num = std::distance(buffer.select<osmium::OSMObject>().cbegin(), buffer.select<osmium::OSMObject>().cend());
+    REQUIRE(num > 0);
+    REQUIRE(buffer.select<osmium::OSMObject>().cbegin()->id() == 1);
+
+    std::string filename;
+
+    SECTION("Successfull writes") {
+
+        SECTION("Writer buffer") {
```

Comment:
Copying the original test was a quick hack.

No problem, I will refactor once we agree what to do about `create_instance` vs ctor.